Purchases/Networking/Operations/GetIntroEligibilityOperation.swift

```diff
@@ -0,0 +1,123 @@
+            }
+
+            var eligibilities: [String: IntroEligibility] = [:]
+
+            for productID in self.productIdentifiers {
+                eligibilities[productID] = IntroEligibility(eligibilityStatus: .unknown)
+            }
+
+            completion(eligibilities, nil)
+            return
+        }
+
+        // Closure we can use for both missing appUserID as well as server error where we have an unknown
+        // eligibility status.
+        let unknownEligibilityClosure: () -> [String: IntroEligibility] = {
+            let unknownEligibilities = [IntroEligibility](repeating: IntroEligibility(eligibilityStatus: .unknown),
+                                                          count: self.productIdentifiers.count)
+            let productIdentifiersToEligibility = zip(self.productIdentifiers, unknownEligibilities)
+            return Dictionary(uniqueKeysWithValues: productIdentifiersToEligibility)
+        }
+
+        guard let appUserID = try? self.configuration.appUserID.escapedOrError() else {
+            self.completion(unknownEligibilityClosure(), ErrorUtils.missingAppUserIDError())
+            return
+        }
+
+        let fetchToken = self.receiptData.asFetchToken
+        let path = "/subscribers/\(appUserID)/intro_eligibility"
+        let body: [String: Any] = ["product_identifiers": self.productIdentifiers,
+                                   "fetch_token": fetchToken]
+
+        httpClient.performPOSTRequest(serially: true,
+                                      path: path,
+                                      requestBody: body,
+                                      headers: authHeaders) { statusCode, maybeResponse, error in
+            let eligibilityResponse = IntroEligibilityResponse(maybeResponse: maybeResponse,
+                                                               statusCode: statusCode,
+                                                               error: error,
+                                                               productIdentifiers: self.productIdentifiers,
+                                                               unknownEligibilityClosure: unknownEligibilityClosure,
+                                                               completion: self.completion)
+            self.handleIntroEligibility(response: eligibilityResponse)
+        }
+    }
+
+    func handleIntroEligibility(response: IntroEligibilityResponse) {
+        var eligibilitiesByProductIdentifier = response.maybeResponse
+        if response.statusCode >= HTTPStatusCodes.redirect.rawValue || response.error != nil {
+            eligibilitiesByProductIdentifier = [:]
+        }
+
+        guard let eligibilitiesByProductIdentifier = eligibilitiesByProductIdentifier else {
+            response.completion(response.unknownEligibilityClosure(), nil)
+            return
+        }
+
+        var eligibilities: [String: IntroEligibility] = [:]
```

Comment:
This could be a bit simpler:
```swift
let eligibilities: [String: IntroEligibilityStatus] = Set(response.productIdentifiers)
  .dictionaryWithValues {
   if let eligibility = eligibilitiesByProductIdentifier[productID] as? Bool {
      return eligibility ? .eligible : .ineligible
    } else {
      return .unknown
    }   
  }
  .mapValues(IntroEligibility.init(eligibilityStatus:))
```

Reply:
Much cleaner